### PR TITLE
Fixed 'label' inconsistency in Instruction's inverse() method

### PR DESCRIPTION
**Pull request by Jwala-1908**
### Summary
Label of the inverse of a gate now becomes gate_label + '_dg' instead of None earlier.

### Details and comments
The inverse of a gate obtained by gate.inverse() did not reflect the label that the gate was initialised with. Now the label of inverse will also carry the required string that the gate had, followed by '_dg'.

Previous:
error

Updated:
corrected

### CODE REVIEW

**Comment by kdk** [comment]
Hi @Jwala-1908 , thanks for opening this PR. What's the motivation for this change? In general, `label` is set by the user and not modified, unlike `name`. (`Instruction.power()` seems like an exception to this ). Maybe @1ucian0 has an idea for the correct behavior.

**Comment by Jwala-1908** [comment]
When the circuits are visualized, the label is supposed to be printed on the gates. So if (as per the above images) we would create a new circuit and append `qc_gate` and `qc_gate_inv` and draw the circuit, the label on `qc_gate` used to be labelled with the assigned label, while it's inverse contained no label and displayed the 'name' attribute of the inverse instead.

Previous:
image

After this change:
image

File: qiskit/circuit/instruction.py [hunk 1]
```diff
@@ -325,9 +325,14 @@ def inverse(self):
                                        params=self.params.copy())
 
         else:
+            if getattr(self, 'label', None) is None:
```
Comment (itoko) [inline qiskit/circuit/instruction.py:328]: `if self.label is None:` ?

**Comment by itoko** [comment]
I personally like the idea carrying `label` by default. Here is a spec table of interest. 
```
print(qc_gate.name)	# 'circuit10'
print(qc_gate.label)	# 'qc_gate'
print(qc_gate.inverse().name)	# 'circuit10_dg'
print(qc_gate.inverse().label)	# None (=> 'qc_gate_dg')
print(qc_gate.inverse().inverse().name)  # 'circuit10_dg_dg' (=> 'circuit10'?)
print(qc_gate.inverse().inverse().label) # None (=> 'qc_gate_dg_dg')
print(qc_gate.power(2).name)	# 'unitary'
print(qc_gate.power(2).label)	# 'circuit10^2'
print(qc_gate.power(2).power(2).name)  # 'unitary'
print(qc_gate.power(2).power(2).label) # 'unitary^2' (=> 'circuit10^2^2')
```
What do you think? @1ucian0

File: qiskit/circuit/instruction.py (hunk 1 above)
Comment (Jwala-1908) [reply qiskit/circuit/instruction.py:328]: Hi @itoko, I tried that before. There are cases where the 'self' calling the inverse is of `Instruction` class, which does not have any `label` attribute. Thus attempting to read `self.label` will throw an error. Instead, by using `getattr` we are checking whether `self` has the attribute `label` before trying to read it.

**Comment by Jwala-1908** [comment]
> I personally like the idea carrying `label` by default. Here is a spec table of interest.
> 
> ```
> print(qc_gate.name)	# 'circuit10'
> print(qc_gate.label)	# 'qc_gate'
> print(qc_gate.inverse().name)	# 'circuit10_dg'
> print(qc_gate.inverse().label)	# None (=> 'qc_gate_dg')
> print(qc_gate.inverse().inverse().name)  # 'circuit10_dg_dg' (=> 'circuit10'?)
> print(qc_gate.inverse().inverse().label) # None (=> 'qc_gate_dg_dg')
> print(qc_gate.power(2).name)	# 'unitary'
> print(qc_gate.power(2).label)	# 'circuit10^2'
> print(qc_gate.power(2).power(2).name)  # 'unitary'
> print(qc_gate.power(2).power(2).label) # 'unitary^2' (=> 'circuit10^2^2')
> ```
> 
> What do you think? @1ucian0

Interesting indeed. Can I work on rectifying this/these as well?

**Comment by 1ucian0** [comment]
I think this is part of the general [name-vs-label general confusion](https://github.com/Qiskit/qiskit-terra/issues/2393). I tend to agree with @kdk that labels [are set for "special" cases](https://github.com/Qiskit/qiskit-terra/pull/5600#issuecomment-758717535). The motivational example works with label:
```
from qiskit import QuantumCircuit
qc = QuantumCircuit(2, name="qc_gate")
qc.cx(0,1)
qc_gate = qc.to_gate()
qc_gate_inv = qc_gate.inverse()

circuit = QuantumCircuit(2)
circuit.append(qc_gate, [0,1])
circuit.append(qc_gate_inv, [0,1])
circuit.draw('mpl')
```
image

**Pull request closed without merging**